src/platform/w3dengine/client/w3ddisplay.cpp

```diff
@@ -1389,47 +1389,49 @@ void W3DDisplay::Preload_Texture_Assets(Utf8String texture)
     }
 }
 
-#define BFT_BITMAP 0x4d42 // BM
-
+void Create_Bmp_File(const char *filename, const char *data, int width, int height)
+{
+#ifdef BUILD_WITH_D3D8
+#define BFT_BITMAP 0x4d42 // 'BM'
 #define DibNumColors(lpbi) \
     ((lpbi)->biClrUsed == 0 && (lpbi)->biBitCount <= 8 ? (int)(1 << (int)(lpbi)->biBitCount) : (int)(lpbi)->biClrUsed)
-
 #define DibSize(lpbi) ((lpbi)->biSize + (lpbi)->biSizeImage + (int)(lpbi)->biClrUsed * sizeof(RGBQUAD))
-
 #define DibPaletteSize(lpbi) (DibNumColors(lpbi) * sizeof(RGBQUAD))
-void Create_Bmp_File(const char *filename, const char *data, int width, int height)
-{
-#ifdef BUILD_WITH_D3D8
-    BITMAPINFOHEADER *header = (BITMAPINFOHEADER *)LocalAlloc(LMEM_ZEROINIT, sizeof(BITMAPINFOHEADER));
-    header->biSize = sizeof(BITMAPINFOHEADER);
-    header->biWidth = width;
-    header->biHeight = height;
-    header->biPlanes = 1;
-    header->biBitCount = 24;
-    header->biCompression = 0;
-    header->biSizeImage = 24 * header->biHeight * ((header->biWidth + 7) / 8);
-    header->biClrImportant = 0;
-    HANDLE h = CreateFile(filename, GENERIC_READ | GENERIC_WRITE, 0, nullptr, CREATE_ALWAYS, FILE_ATTRIBUTE_NORMAL, nullptr);
+
+    // #BUGFIX Simplify code and avoid LocalAlloc. Use the stack.
+    BITMAPINFOHEADER iheader = { 0 };
+    iheader.biSize = sizeof(BITMAPINFOHEADER);
+    iheader.biWidth = width;
+    iheader.biHeight = height;
+    iheader.biPlanes = 1;
+    iheader.biBitCount = 24;
+    iheader.biCompression = 0;
+    iheader.biSizeImage = 24 * iheader.biHeight * ((iheader.biWidth + 7) / 8);
```

Comment:
why can this not just be 3 * height * width?

Reply:
Excellent point. I didn't think twice about the weird math. `3 * height * width` yields same result. Tested and worked.